### PR TITLE
update document builder.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @superyyrrzz @qinezh @vicancy @hellosnow @ansyral

### CODE REVIEW

**Comment by vicancy** [comment]
why clone is needed?

**Comment by vwxyzh** [comment]
Because `ToList()`

File: src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs [hunk 1]
```diff
@@ -24,6 +24,7 @@ public class SingleDocumentBuilder : IDisposable
         public CompositionHost Container { get; set; }
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs]: remove?

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuildParameters.cs [hunk 2]
```diff
@@ -52,5 +52,8 @@ public sealed class DocumentBuildParameters : MarshalByRefObject
 
         [IncrementalIgnore]
         public bool ForceRebuild { get; set; }
+
+        public DocumentBuildParameters Clone() =>
+            (DocumentBuildParameters)MemberwiseClone();
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuildParameters.cs]: do we need deep copy?

File: src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs [hunk 3]
```diff
@@ -78,8 +78,27 @@ public void Build(DocumentBuildParameters parameter)
...
+            if (parameters == null)
+            {
+                throw new ArgumentNullException(nameof(parameters));
+            }
+            if (parameters.Count == 0)
+            {
+                throw new ArgumentException("Parameters is empty.", nameof(parameters));
+            }
+
+            var markdownServiceProvider = (IMarkdownServiceProvider)_container.GetExport(
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs]: >_container.GetExport [](start = 68, length = 20)

can call GetExport method in this class

**Comment by hellosnow** [comment]
:shipit:

**Pull request merged**